server/service/integration_mdm_test.go

```diff
@@ -587,6 +589,179 @@ func (s *integrationMDMTestSuite) TestMDMAppleUnenroll() {
+		NodeKey:         ptr.String(t.Name()),
+		UUID:            uuid.New().String(),
+		Hostname:        fmt.Sprintf("%sfoo.local", t.Name()),
+		Platform:        "darwin",
+	})
+	require.NoError(t, err)
+
+	// add an encryption key for the host
+	cert, _, _, err := s.fleetCfg.MDM.AppleSCEP()
+	require.NoError(t, err)
+	parsed, err := x509.ParseCertificate(cert.Certificate[0])
+	require.NoError(t, err)
+	recoveryKey := "AAA-BBB-CCC"
+	encryptedKey, err := pkcs7.Encrypt([]byte(recoveryKey), []*x509.Certificate{parsed})
+	require.NoError(t, err)
+	base64EncryptedKey := base64.StdEncoding.EncodeToString(encryptedKey)
+
+	err = s.ds.SetOrUpdateHostDiskEncryptionKey(ctx, host.ID, base64EncryptedKey)
+	require.NoError(t, err)
+
+	// request with no token
+	res := s.DoRawNoAuth("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusUnauthorized)
+	res.Body.Close()
+
+	// encryption key not procesed yet
+	resp := getHostEncryptionKeyResponse{}
+	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusNotFound, &resp)
+
+	// unable to decrypt encryption key
+	err = s.ds.SetHostDiskEncryptionKeyStatus(ctx, []uint{host.ID}, false)
+	require.NoError(t, err)
+	resp = getHostEncryptionKeyResponse{}
+	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusNotFound, &resp)
+
+	// no activities created so far
+	activities := listActivitiesResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/activities", nil, http.StatusOK, &activities)
+	found := false
+	for _, activity := range activities.Activities {
+		if activity.Type == "read_host_disk_encryption_key" {
+			found = true
+		}
+	}
+	require.False(t, found)
+
+	// decryptable key
+	checkDecryptableKey := func(u fleet.User) {
+		err = s.ds.SetHostDiskEncryptionKeyStatus(ctx, []uint{host.ID}, true)
+		require.NoError(t, err)
+		resp = getHostEncryptionKeyResponse{}
+		s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/encryption_key", host.ID), nil, http.StatusOK, &resp)
+		require.Equal(t, recoveryKey, resp.EncryptionKey.DecryptedValue)
+
+		// use the admin token to get the activities
+		currToken := s.token
+		defer func() { s.token = currToken }()
+		s.token = s.getTestAdminToken()
+		activities = listActivitiesResponse{}
+		s.DoJSON("GET", "/api/latest/fleet/activities", nil, http.StatusOK, &activities, "order_key", "created_at")
+		found = false
+		for _, activity := range activities.Activities {
+			if activity.Type == "read_host_disk_encryption_key" && u.ID == *activity.ActorID {
+				found = true
+				require.JSONEq(t, fmt.Sprintf(`{"host_display_name": "%s"}`, host.DisplayName()), string(*activity.Details))
+			}
+		}
+		require.True(t, found)
+	}
+
+	// we're about to mess up with the token, make sure to set it to the
+	// default value when the test ends
+	currToken := s.token
+	t.Cleanup(func() { s.token = currToken })
+
+	// admins are able to see the token
```

Comment:
Do you mean "admins are able to see the host encryption key"?

Reply:
yes! thank you 🙌